airbyte-integrations/connectors/source-jdbc/src/main/java/io/airbyte/integrations/source/jdbc/JdbcCdcStateManager.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2020 Airbyte
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package io.airbyte.integrations.source.jdbc;
+
+import com.google.common.annotations.VisibleForTesting;
+import io.airbyte.integrations.source.jdbc.models.CdcState;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class JdbcCdcStateManager {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(JdbcStateManager.class);
+
+  private final CdcState initialSerialized;
+  private final AirbyteFileOffsetBackingStore offsetStore;
+
+  public JdbcCdcStateManager(CdcState serialized) {
+    this(serialized, new AirbyteFileOffsetBackingStore());
+  }
+
+  @VisibleForTesting
+  JdbcCdcStateManager(CdcState serialized, AirbyteFileOffsetBackingStore offsetStore) {
+    this.offsetStore = offsetStore;
+    this.initialSerialized = serialized;
+
+    LOGGER.info("Initialized CDC state with: {}", serialized);
+    initializeOffsetStore(initialSerialized != null ? initialSerialized : null);
+  }
+
+  public CdcState getCdcState() {
+    return offsetStore.read();
+  }
+
+  public void initializeOffsetStore(CdcState cdcState) {
```

Comment:
`persistCdcState` seems like it'd make more sense

Reply:
removed in the rearrangement.